torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6068,23 +6068,23 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            op=torch.lu,
            dtypes=floating_and_complex_types(),
            supports_inplace_autograd=False,
+           # we use in-place operations which cannot be avoided.
+           # This causes vmap failures, hence we skip batched gradient checks
+           check_batched_grad=False,
            check_batched_gradgrad=False,
            supports_out=False,
            sample_inputs_func=sample_inputs_lu,
            decorators=[skipCUDAIfNoMagmaAndNoCusolver, skipCUDAIfRocm, skipCPUIfNoLapack],
            skips=(
-               # we skip jit tests because lu_backward is impelemented as autograd.Function,
-               # which does not support autograd with scripting
+               # we skip jit tests because `lu` is a torch function
```

Comment:
I don't understand what you mean by this. `torch.lu` is already being tested for JIT here: https://github.com/pytorch/pytorch/blob/48af9de92ffae89618a22291738189e3110d14ed/test/test_jit.py#L8904-L8906 so it seems like it does work with TorchScript right?

Reply:
yes, but test_jit_consistency fails, and, apparently, it is related to the CPP bindings, but I could be wrong.

Reply:
can you paste the error here?

Reply:
@anjali411 , there are the errors:
```python
___________________________________________ TestJitCPU.test_variant_consistency_jit_lu_cpu_complex64 ____________________________________________Traceback (most recent call last):                                                                                                                
  File "/home/nik/git/Quansight/pytorch/test/test_ops.py", line 713, in test_variant_consistency_jit                                              
    no_grad=not _requires_grad, no_gradgrad=not op.supports_gradgrad)                                                                             
  File "/home/nik/git/Quansight/pytorch/torch/testing/_internal/common_jit.py", line 90, in check_against_reference                               
    outputs_test = self.runAndSaveRNG(func, nograd_inputs, kwargs)                                                                                
  File "/home/nik/git/Quansight/pytorch/torch/testing/_internal/common_jit.py", line 160, in runAndSaveRNG                                        
    results = func(*inputs, **kwargs)                                                                                                             
  File "/home/nik/git/Quansight/pytorch/torch/testing/_internal/jit_metaprogramming_utils.py", line 307, in script_fn                             
    fn, tensors = gen_script_fn_and_args(method_name, func_type, *args, **kwargs)                                                                 
  File "/home/nik/git/Quansight/pytorch/torch/testing/_internal/jit_metaprogramming_utils.py", line 298, in gen_script_fn_and_args                
    CU = torch.jit.CompilationUnit(script)                                                                                                        
RuntimeError:                                                                                                                                     
'Tensor (inferred)' object has no attribute or method 'lu'.:                                                                                      
  File "<string>", line 3                                                                                                                         
                                                                                                                                                  
def the_method(i0):                                                                                                                               
    return i0.lu(True, True)                                                                                                                      
           ~~~~~ <--- HERE                                                                                                                        
                                                                                                                                                  
_____________________________________________ TestJitCPU.test_variant_consistency_jit_lu_cpu_float32 _____________________________________________
Traceback (most recent call last):                                                                                                                
  File "/home/nik/git/Quansight/pytorch/test/test_ops.py", line 731, in test_variant_consistency_jit                                              
    func_type=func_type, aten_name=op.aten_name)                                                                                                  
  File "/home/nik/git/Quansight/pytorch/torch/testing/_internal/jit_metaprogramming_utils.py", line 490, in check_alias_annotation                
    torch._C._jit_check_alias_annotation(CU.the_method.graph, tuple(tensors), aten_name)                                                          
RuntimeError: aliasOp != torch::jit::getOperatorAliasMap().end()INTERNAL ASSERT FAILED at "../torch/csrc/jit/passes/utils/check_alias_annotation.c
pp":159, please report a bug to PyTorch.                       
```

Reply:
We should add this particular error message to the comment. This is not a regression, though. Here's my results on Colab:

```
def foo(t):
  t.lu()
scripted_foo = torch.jit.script(foo)
: RuntimeError: 'Tensor (inferred)' object has no attribute or method 'lu'.:
```

@nikitaved Would you file an issue for the `lu` method variant not working when scripting?